manager/allocator/cnmallocator/drivers_ipam.go

```diff
@@ -1,14 +1,39 @@
 package cnmallocator
 
 import (
+	"net"
+	"strconv"
+	"strings"
+
 	"github.com/docker/libnetwork/drvregistry"
 	"github.com/docker/libnetwork/ipamapi"
 	builtinIpam "github.com/docker/libnetwork/ipams/builtin"
 	nullIpam "github.com/docker/libnetwork/ipams/null"
 	remoteIpam "github.com/docker/libnetwork/ipams/remote"
+	"github.com/docker/libnetwork/ipamutils"
+	"github.com/sirupsen/logrus"
 )
 
-func initIPAMDrivers(r *drvregistry.DrvRegistry) error {
+func initIPAMDrivers(r *drvregistry.DrvRegistry, defaultAddrPool []*net.IPNet, subnetSize int) error {
+	var addressPool []*ipamutils.NetworkToSplit
+	var str strings.Builder
+	str.WriteString("Subnetlist - ")
+	if defaultAddrPool != nil {
+		for _, p := range defaultAddrPool {
+			// trim leading and trailing white spaces
+			poolsDef := ipamutils.NetworkToSplit{}
+			poolsDef.Base = p.String()
+			poolsDef.Size = subnetSize
+			addressPool = append(addressPool, &poolsDef)
+			str.WriteString(p.String() + ",")
+		}
+		str.WriteString(": Size ")
+		str.WriteString(strconv.Itoa(subnetSize))
+	}
+	ipamutils.ConfigGlobalScopeDefaultNetworks(addressPool)
```

Comment:
it doesn't make much sense to me to have IPNet objects and having to convert them to strings to just initialize the ipam. This needs to be cleaned. Use the object